PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'account.move'
-    _inherit = 'account.move.line'
-    _inherit = 'fleet.vehicle'
-    _inherit = 'fleet.vehicle.log.services'

Total Changes: 9

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/account_fleet/models/account_move.py
+++ b/addons/account_fleet/models/account_move.py
@@ -4,7 +4,7 @@ from odoo import fields, models, _
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     def _post(self, soft=True):
         vendor_bill_service = self.env.ref('account_fleet.data_fleet_service_type_vendor_bill', raise_if_not_found=False)
@@ -30,7 +30,7 @@ class AccountMove(models.Model):
 
 
 class AccountMoveLine(models.Model):
-    _inherit = 'account.move.line'
+    _inherit = ['account.move.line']
 
     vehicle_id = fields.Many2one('fleet.vehicle', string='Vehicle', index='btree_not_null')
     # used to decide whether the vehicle_id field is editable

--- a/addons/account_fleet/models/fleet_vehicle.py
+++ b/addons/account_fleet/models/fleet_vehicle.py
@@ -5,7 +5,7 @@ from odoo import Command, models, fields
 
 
 class FleetVehicle(models.Model):
-    _inherit = 'fleet.vehicle'
+    _inherit = ['fleet.vehicle']
 
     bill_count = fields.Integer(compute='_compute_move_ids', string="Bills Count")
     account_move_ids = fields.One2many('account.move', compute='_compute_move_ids')

--- a/addons/account_fleet/models/fleet_vehicle_log_services.py
+++ b/addons/account_fleet/models/fleet_vehicle_log_services.py
@@ -3,8 +3,9 @@
 from odoo import api, fields, models, _
 from odoo.exceptions import UserError
 
+
 class FleetVehicleLogServices(models.Model):
-    _inherit = 'fleet.vehicle.log.services'
+    _inherit = ['fleet.vehicle.log.services']
 
     account_move_line_id = fields.Many2one(comodel_name='account.move.line')  # One2one
     account_move_state = fields.Selection(related='account_move_line_id.parent_state')
